src/absil/illib.fs

```diff
@@ -472,7 +473,11 @@ module ResizeArray =
     let mapToSmallArrayChunks f (inp: ResizeArray<'t>) =
         let itemSizeBytes = sizeof<'t>
         // rounding down here is good because it ensures we don't go over
-        let maxArrayItemCount = LOH_SIZE_THRESHOLD_BYTES / itemSizeBytes
+        let maxArrayItemCount =
+            if itemSizeBytes < REASONABLE_THRESHOLD_TO_AVOID_LOH then
+                REASONABLE_THRESHOLD_TO_AVOID_LOH / itemSizeBytes
+            else
+                REASONABLE_THRESHOLD_TO_AVOID_LOH
```

Comment:
Wondering if this is too complex and should just be 1024

Reply:
;-)